pilot/pkg/model/service.go

```diff
@@ -84,6 +86,13 @@ type Service struct {
 	// or use the passthrough model (i.e. proxy will forward the traffic to the network endpoint requested
 	// by the caller)
 	Resolution Resolution
+
+	// ID represents the location of the config used to create this Service. Optional.
+	// For k8s, it is namespace:name
+	ID string `json:"name,omitempty"`
```

Comment:
If you used `ConfigMeta` for all of this data then [`ConfigMeta.Key()`](https://godoc.org/github.com/istio/istio/pilot/pkg/model#ConfigMeta.Key) would replace the need for this field.

Reply:
That's adding a k8s dependency, I agree with Shriram we don't want that.